c_glib/arrow-glib/compute.cpp

```diff
@@ -327,15 +313,23 @@ garrow_execute_context_class_init(GArrowExecuteContextClass *klass)
 
 /**
  * garrow_execute_context_new:
+ * @thread_pool: (nullable): A #GArrowThreadPool or %NULL.
  *
  * Returns: A newly created #GArrowExecuteContext.
  *
  * Since: 1.0.0
  */
 GArrowExecuteContext *
-garrow_execute_context_new(void)
+garrow_execute_context_new(GArrowThreadPool *thread_pool)
 {
   auto execute_context = g_object_new(GARROW_TYPE_EXECUTE_CONTEXT, NULL);
+  auto priv = GARROW_EXECUTE_CONTEXT_GET_PRIVATE(execute_context);
+
+  auto arrow_thread_pool = garrow_thread_pool_get_raw(thread_pool);
+  priv->thread_pool = arrow_thread_pool;
```

Comment:
We should set `thread_pool` as property.

Reply:
I changed it, but that means we cannot create the `arrow::compute::ExecContext` until the property is set. Is that ok?

Reply:
Ok, updated.